fsharp-backend/src/BackendOnlyStdLib/LibDarkInternal.fs

```diff
@@ -765,6 +765,85 @@ human-readable data."
       deprecated = NotDeprecated }
 
 
+    { name = fn "DarkInternal" "startStaticAssetDeploy" 0
+      parameters = [ Param.make "username" TStr ""; Param.make "canvasID" TUuid "" ]
+      returnType = TResult(TStr, TStr)
+      description =
+        "Records an in-progress static asset deployment, returning the deployHash"
+      fn =
+        internalFn (function
+          | _, [ DStr username; DUuid canvasID ] ->
+            uply {
+              match! Account.getUser (UserName.create username) with
+              | None -> return DResult(Error(DStr "User not found"))
+              | Some user ->
+                let! canvasMeta = Canvas.getMetaFromID canvasID
+                let! deployHash =
+                  StaticAssets.startStaticAssetDeploy user canvasID canvasMeta.name
+                return DResult(Ok(DStr deployHash))
+            }
+          | _ -> incorrectArgs ())
+      sqlSpec = NotQueryable
+      previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "DarkInternal" "finishStaticAssetDeploy" 0
+      parameters =
+        [ Param.make "canvasID" TUuid ""; Param.make "deployHash" TStr "" ]
+      returnType =
+        TResult(
+          TRecord [ "deployHash", TStr
+                    "url", TStr
+                    "status", TStr
+                    "lastUpdate", TDate ],
+          TStr
+        )
+      description = "Marks an in-progress static asset deployment as finished"
+      fn =
+        internalFn (function
+          | _, [ DUuid canvasID; DStr deployHash ] ->
+            uply {
+              let! canvasMeta = Canvas.getMetaFromID canvasID
+              let! deploy =
+                StaticAssets.finishStaticAssetDeploy
+                  canvasID
+                  canvasMeta.name
+                  deployHash
+              return
+                DObj(
+                  Map [ "deployHash", DStr deploy.deployHash
```

Comment:
Similar to `start`, what's the reason for this structure? Most of it is known to the caller already, right?

Reply:
Now returning a plain Result<Null, string>. We could return the update date, but I don't currently feel a need. We can always ajust.